spring-cloud-stream/src/main/java/org/springframework/cloud/stream/aggregate/NamespaceAwareSpringIntegrationMetricReader.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.aggregate;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.springframework.boot.actuate.metrics.Metric;
+import org.springframework.boot.actuate.metrics.reader.MetricReader;
+import org.springframework.integration.monitor.IntegrationMBeanExporter;
+import org.springframework.integration.support.management.Statistics;
+import org.springframework.util.Assert;
+
+/**
+ * A customized version of {@link org.springframework.boot.actuate.metrics.integration.SpringIntegrationMetricReader} that
+ * provides support for customizing channels with a namespace prefix.
+ *
+ * @author Marius Bogoevici
+ * @see org.springframework.boot.actuate.metrics.integration.SpringIntegrationMetricReader for original implementation
+ */
+public class NamespaceAwareSpringIntegrationMetricReader implements MetricReader {
+
+	private final String namespace;
+
+	private final IntegrationMBeanExporter exporter;
+
+	public NamespaceAwareSpringIntegrationMetricReader(String namespace, IntegrationMBeanExporter exporter) {
+		Assert.hasText(namespace, "cannot be null or empty String");
+		Assert.notNull(exporter, "cannot be null");
+		this.namespace = namespace;
+		this.exporter = exporter;
+	}
+
+	@Override
+	public Metric<?> findOne(String metricName) {
+		return null;
+	}
+
+	@Override
+	public Iterable<Metric<?>> findAll() {
+		IntegrationMBeanExporter exporter = this.exporter;
```

Comment:
No-op. I think that was just blink copy-paste: https://github.com/spring-projects/spring-boot/blob/master/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/integration/SpringIntegrationMetricReader.java#L49 :smile: .

Here is the original change: https://github.com/spring-projects/spring-boot/commit/31d6a0f17a66e5342b9d2337cffdbcd3c6f2b6cd.
I guess that was just quick fix of `RequireThis` Checkstyle rule violation :smile: